snuba/datasets/schema.py

```diff
@@ -8,46 +9,40 @@ def local_dataset_mode():
     return settings.DATASET_MODE == "local"
 
 
-class Schema(object):
+class Schema(ABC):
     """
-    Represents the full set of columns in a clickhouse table, this only contains
-    basic metadata for now.
+    A schema is an abstraction over a data model we can query.
+    It provides a set of columns and a where clause to build the query.
+    Concretely this can represent a table, a view or a group of
+    joined tables.
+    This level of abstraciton only provides read primitives.
+
+    Going on this will be abstract from Clickhouse details. As of
+    now this distinction does not exists to this provides CLickhouse
+    specific details.
     """
 
-    TEST_TABLE_PREFIX = "test_"
-
-    def __init__(self, local_table_name, dist_table_name, columns, migration_function=None):
+    def __init__(
+        self,
+        columns: ColumnSet,
+        migration_function: Optional[Callable[[str, Mapping[str, str]], Sequence[str]]]=None,
+    ) -> None:
         self.__columns = columns
+        self.__migration_function = migration_function if migration_function else lambda table, schema: []
```

Comment:
If this is optional and `get_migration_statements` is also typed as `Optional`, why does this have a default to fit the non-optional type constraint?